zio-kafka/src/main/scala/zio/kafka/consumer/internal/RunloopAccess.scala

```diff
@@ -55,16 +55,23 @@ private[consumer] final class RunloopAccess private (
    */
   def subscribe(
     subscription: Subscription
-  ): ZIO[Scope, InvalidSubscriptionUnion, UStream[Take[Throwable, PartitionAssignment]]] =
+  ): ZIO[Scope, InvalidSubscriptionUnion, SubscriptionStreamControl[UStream[Take[Throwable, PartitionAssignment]]]] =
     for {
-      stream <- ZStream.fromHubScoped(partitionHub)
+      partitionAssignmentQueue <- Queue.unbounded[Take[Throwable, PartitionAssignment]]
+      stream                   <- ZStream.fromHubScoped(partitionHub)
+      _                        <- stream.flattenTake.runIntoQueue(partitionAssignmentQueue).forkScoped
       // starts the Runloop if not already started
       _ <- withRunloopZIO(requireRunning = true)(_.addSubscription(subscription))
       _ <- ZIO.addFinalizer {
              withRunloopZIO(requireRunning = false)(_.removeSubscription(subscription)) <*
                diagnostics.emit(Finalization.SubscriptionFinalized)
            }
-    } yield stream
+    } yield SubscriptionStreamControl(
+      ZStream.fromQueue(partitionAssignmentQueue),
+      withRunloopZIO(requireRunning = true)(_.stopSubscribedTopicPartitions(subscription)) *> partitionAssignmentQueue
```

Comment:
```suggestion
      withRunloopZIO(requireRunning = false)(_.stopSubscribedTopicPartitions(subscription)) *> partitionAssignmentQueue
```

Not sure you want to use `true` here. `true` means: if the runloop is not running, start it and apply the `stopSubscribedTopicPartitions` function. 
In your case, IMO, if the Runloop isn't running, calling the `control.stop` should be a no-op, which will be the case if you use `false` instead

Reply:
> In your case, IMO, if the Runloop isn't running, calling the `control.stop` should be a no-op

We need to stop the subscription also when runloop isn't running. Right?